src/storage/lock_manager/deadlock.rs

```diff
@@ -22,91 +22,158 @@ use std::fmt::{self, Display, Formatter};
 use std::rc::Rc;
 use std::sync::Arc;
 use std::time::{Duration, Instant};
+use tikv_util::time;
 use tokio_core::reactor::Handle;
 use tokio_timer::Interval;
 
 // 2 mins
 const TXN_DETECT_INFO_TTL: u64 = 120000;
 
-#[derive(Default)]
-struct DetectTable {
-    // txn_ts => (lock_ts, Vec<lock_hash>)
-    wait_for_map: HashMap<u64, HashMap<u64, Vec<u64>>>,
+/// `Locks` is a set of locks belonging to one transaction.
+struct Locks {
+    ts: u64,
```

Comment:
Could we remove it?

Reply:
Yes, it's useless. But it's hard to name it without ts.

Reply:
I prefer to keep it.